sources/packages/web/src/views/student/NoticeOfAssessment.vue

```diff
@@ -41,20 +64,67 @@ export default {
       required: true,
     },
   },
-  setup(props: any) {
+  setup(props) {
+    const router = useRouter();
+    const cancelApplicationModal = ref({} as ModalDialog<boolean>);
+    const assessment = ref<AssessmentNOAAPIOutDTO>();
     const snackBar = useSnackBar();
+    const setAssessment = async () => {
+      assessment.value =
+        await StudentAssessmentsService.shared.getAssessmentNOA(
+          props.assessmentId,
+        );
+    };
     const confirmAssessment = async () => {
       try {
         await StudentAssessmentsService.shared.confirmAssessmentNOA(
           props.assessmentId,
         );
+        await setAssessment();
         snackBar.success("Confirmation of Assessment completed successfully!");
       } catch (error) {
         snackBar.error("An error happened while confirming the assessment.");
       }
     };
 
-    return { confirmAssessment, StudentRoutesConst };
+    watch(
+      () => props.assessmentId,
+      async () => {
+        await setAssessment();
+      },
+      { immediate: true },
+    );
+
+    const viewOnly = computed(
+      () =>
+        !(
+          assessment.value?.applicationStatus ===
+            ApplicationStatus.assessment &&
+          assessment.value?.noaApprovalStatus === AssessmentStatus.required
+        ),
+    );
+
+    const confirmCancelApplication = async () => {
+      if (await cancelApplicationModal.value.showModal(props.applicationId)) {
```

Comment:
As I mentioned in the previous comment, even if cancel application API fails, it will return true here.